### DISCUSSION TITLE
`serverVersion` requirement in the DATABASE_URL?

### DISCUSSION THREAD

**Discussion by plaes** [symfony/symfony#58491] [Q&A] [answered]
Why is `serverVersion` variable required in the `DATABASE_URL`?
DBAL doesn't really need it ([it's optional there](https://github.com/doctrine/dbal/blob/7a8252418689feb860ea8dfeab66d64a56a64df8/src/DriverManager.php#L63)) but its requirement seems to originate from Symfony's cache module.

**Comment by stof** [comment, marked as answer]
DBAL needs it to be able to get the platform object without actually connecting to the database server. When the server version is not provided in the settings, DBAL will initiate the connection to read the server version from the server to know which platform class it should use.

As a cache warmup requires the database platform (for instance, finalizing the mapping for doctrine/orm depends on knowing the platform as some settings have smart defaults that depend on the platform being used), there is 2 solutions to make it possible to warmup the Symfony cache:

1. define the server version in the settings so that connecting to the DB server is not needed
2. ensure that your `cache:warmup` (or `cache:clear` without the `--no-warmup` flag) runs in on a server where the `DATABASE_URL` value is defined and allows connecting to the database

The second solution is often not viable because of the last part of the condition. For instance, for a docker-based environment, the cache warmup will happen during the build of the image, at a time the database container does not even exist yet.
